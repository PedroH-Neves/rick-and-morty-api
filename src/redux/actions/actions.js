export const FETCH_CHARACTERS_REQUEST = 'FETCH_CHARACTERS_REQUEST';
export const FETCH_CHARACTERS_SUCCESS = 'FETCH_CHARACTERS_SUCCESS';
export const FETCH_CHARACTERS_FAILURE = 'FETCH_CHARACTERS_FAILURE';
export const ADD_FILTER = 'ADD_FILTER';
export const REMOVE_FILTER = 'REMOVE_FILTER';


export const fetchCharactersRequest = () => ({type:FETCH_CHARACTERS_REQUEST});
export const fetchCharactersSuccess = characters => ({type:FETCH_CHARACTERS_SUCCESS, payload: characters});
export const fetchCharactersFailure = error => ({type: FETCH_CHARACTERS_FAILURE, payload: error});
export const addFilter = filter => ({ type: ADD_FILTER, payload: filter });
export const removeFilter = filter => ({ type: REMOVE_FILTER, payload: filter });
