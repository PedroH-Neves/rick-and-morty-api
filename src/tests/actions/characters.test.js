import configureStore from 'redux-mock-store';
import { fetchCharactersRequest, fetchCharactersSuccess, fetchCharactersFailure } from '../../redux/actions/actions';

const mockStore = configureStore();
const store = mockStore({});

describe('characters creation', () => {
  it('should request characters', () => {
    store.dispatch(fetchCharactersRequest(
      {
        loading: true,
        characters: {},
        error: '',
      },
    ));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should fetch characters with success', () => {
    store.dispatch(
      fetchCharactersSuccess({
        loading: false,
        characters: {
          name: 'test character',
        },
        error: '',
      }),
    );
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should fetch characters with failure', () => {
    store.dispatch(
      fetchCharactersFailure({
        loading: false,
        characters: {},
        error: 'some error',
      }),
    );
    expect(store.getActions()).toMatchSnapshot();
  });
});
