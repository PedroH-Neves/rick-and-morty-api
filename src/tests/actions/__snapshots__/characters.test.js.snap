// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`characters creation should fetch characters with failure 1`] = `
Array [
  Object {
    "type": "FETCH_CHARACTERS_REQUEST",
  },
  Object {
    "payload": Object {
      "characters": Object {
        "name": "test character",
      },
      "error": "",
      "loading": false,
    },
    "type": "FETCH_CHARACTERS_SUCCESS",
  },
  Object {
    "payload": Object {
      "characters": Object {},
      "error": "some error",
      "loading": false,
    },
    "type": "FETCH_CHARACTERS_FAILURE",
  },
]
`;

exports[`characters creation should fetch characters with success 1`] = `
Array [
  Object {
    "type": "FETCH_CHARACTERS_REQUEST",
  },
  Object {
    "payload": Object {
      "characters": Object {
        "name": "test character",
      },
      "error": "",
      "loading": false,
    },
    "type": "FETCH_CHARACTERS_SUCCESS",
  },
]
`;

exports[`characters creation should request characters 1`] = `
Array [
  Object {
    "type": "FETCH_CHARACTERS_REQUEST",
  },
]
`;
